lib/call-cabal-project-to-nix.nix

```
@@ -6,7 +6,10 @@
 , plan-sha256   ? null # The hash of the plan-to-nix output (makes the plan-to-nix step a fixed output derivation)
 , materialized  ? null # Location of a materialized copy of the nix files
 , checkMaterialization ? null # If true the nix files will be generated used to check plan-sha256 and material
-, cabalProject  ? null # Cabal project file (when null uses "${src}/cabal.project")
+, cabalProjectFileName ? "cabal.project"
+, cabalProject  ? null # Contents of cabal.project file (when null uses "${src}/cabal.project")
+, cabalProjectLocal ? null  # Contents of cabal.project.local file (when null uses "${src}/cabal.project.local")
```

do we need to default `null`?
```suggestion
, cabalProjectLocal ? "${src}/cabal.project.local"  # Contents of cabal.project.local file
```